internal/js/modules/k6/browser/common/page.go

```diff
@@ -1870,6 +1790,53 @@ func (p *Page) WaitForURL(urlPattern string, opts *FrameWaitForURLOptions, rm Re
 	return nil
 }
 
+// waitForEvent subscribes to the given event and resolves once the predicate
+// determines an event should complete the wait or terminate with an error.
+func (p *Page) waitForEvent(
+	ctx context.Context,
+	eventName PageEventName,
+	predicate func(PageEvent) (bool, error),
+) (_ PageEvent, rerr error) {
+	defer func() {
+		if rerr != nil {
+			rerr = fmt.Errorf("waiting for page %s event: %w", eventName, rerr)
+		}
+	}()
+
+	type pageEventWaitResult struct {
+		event PageEvent
+		err   error
+	}
+
+	var (
+		once   sync.Once
+		result = make(chan pageEventWaitResult, 1)
```

Comment:
Why does it need to be buffered?

Reply:
It avoids goroutine leaks: In case `Context` gets done before receiving from `results`.

Reply:
Ok, i think i understand now. Couldn't you do something like this to avoid the need for a buffered channel. It's usually a sign that a buffered channel is hiding a race condition:

```go
once.Do(func() { 
  select {
  case result <- pageEventWaitResult{event: event, err: perr}:
  case <-ctx.Done():
    perr = ctx.Err()
  }
})
```

Reply:
That's also an option, but practically the same? How would this buffered channel create a race condition?